controller.py

```diff
@@ -322,6 +325,8 @@ def validate_email(self):
         # Manually validate an email address, without the admin having to click on a confirmation link
         uuid = flask.request.form.get("uuid")
         library = self.library_for_request(uuid)
+        if isinstance(library, ProblemDetail):
+            return library
```

Comment:
This won't come up a lot and doesn't relate to the Python 3 conversion, but a problem _inside_ library_for_request was causing it to return a ProblemDetail, and then this method wasn't handling the ProblemDetail properly. I fixed it and added a test.